python/triton/language/semantic.py

```diff
@@ -476,10 +476,16 @@ def not_equal(input: tl.tensor,
 
 
 def arange(start: int, end: int, builder: ir.builder) -> tl.tensor:
-    if not isinstance(start, int) or not isinstance(end, int):
+    if not isinstance(start, int) or not isinstance(end, int) or start >> 32 or end >> 32:
         raise ValueError("arange's arguments must be of type tl.constexpr")
-
-    shape = [end - start]
+    if end <= start:
+        raise ValueError("arange's end argument must be greater than the start argument")
+    TRITON_MAX_TENSOR_NUMEL = 131072
```

Comment:
this should be in the constructor of `tl.block_type`

Reply:
done